### PR TITLE
Upgrade deps, add tools to code to pin version, change to use golangci

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ language: go
 go_import_path: contrib.go.opencensus.io
 
 go:
-  - 1.11.x
+  - 1.15.x
```
Comment (bogdandrutu) [inline .travis.yml:6]: go 1.15 is required only for lint, so the go.mod is still not changed

File: prometheus.go [hunk 2]
```diff
@@ -151,11 +150,12 @@ func newCollector(opts Options, registrar prometheus.Registerer) *collector {
 
 func (c *collector) toDesc(metric *metricdata.Metric) *prometheus.Desc {
 	var labels prometheus.Labels
-	if metric.Resource == nil {
+	switch {
```
Comment (punya) [inline prometheus.go:153]: Was this change made to comply with golangci-lint or as a matter of taste?

(I'm okay with the change, but I'd like to know either way.)

Comment (bogdandrutu) [reply prometheus.go:153]: Yes one of the linters suggested this. Not doing this kind of changes by myself :))

Comment (bogdandrutu) [reply prometheus.go:153]: gocritic is the name of the linters that I enabled and required this. We use exactly the same config in the otel collector

**Comment by bogdandrutu** [comment]
@punya I pushed a second commit which splits internal/tools in its own module so tools are not polluting the main go mod/sum

**Review by punya — APPROVED** [review]

**Pull request merged**